cli/cmd/routes.go

```diff
@@ -69,8 +73,10 @@ This command will only work for services that have a Service Profile defined.`,
 
 	cmd.PersistentFlags().StringVarP(&options.namespace, "namespace", "n", options.namespace, "Namespace of the specified resource")
 	cmd.PersistentFlags().StringVarP(&options.timeWindow, "time-window", "t", options.timeWindow, "Stat window (for example: \"10s\", \"1m\", \"10m\", \"1h\")")
-	cmd.PersistentFlags().StringVar(&options.fromResource, "from", options.fromResource, "If present, restricts outbound stats from the specified resource name")
-	cmd.PersistentFlags().StringVar(&options.fromNamespace, "from-namespace", options.fromNamespace, "Sets the namespace used from lookup the \"--from\" resource; by default the current \"--namespace\" is used")
+	cmd.PersistentFlags().StringVar(&options.toService, "to-service", options.toService, "If present, shows outbound stats to the specified service name")
+	cmd.PersistentFlags().StringVar(&options.toServiceNamespace, "to-service-namespace", options.toServiceNamespace, "Sets the namespace used to lookup the \"--to-service\" resource; by default the current \"--namespace\" is used")
+	cmd.PersistentFlags().BoolVar(&options.toAll, "to-all", options.toAll, "If present, shows outbound stats to all services")
+	cmd.PersistentFlags().StringVar(&options.toExternal, "to-external", options.toExternal, "If present, shows outbound stats to the specified external DNS name")
```

Comment:
This new set of flags doesn't feel congruous with the `linkerd stat` command. My preference would be to use `--to` and `--to-namespace` flags, similar to how stat is handled.

## Examples

To query route stats for requests from the webapp deployment to the books service:

```
linkerd routes deploy/webapp --to svc/books
```

To query route stats for requests from all deployments to all services:

```
linkerd routes deploy --to svc
```

To query route stats for requests to the "linkerd.io" domain:

```
linkerd routes ns/default --to au/linkerd.io
```

Reply:
My thinking is that `--to` can ONLY be an authority (or an authority constructed from a service).  By encoding this restriction into the type of the flag, we avoid invalid inputs like `--to deploy/foo`.  We can gate against those cases with input validation, but I had hoped that naming the flags this way made it more immediately clear what inputs were valid.